cpp/include/raft/sparse/linalg/spmm.hpp

```diff
@@ -60,11 +60,16 @@ void spmm(raft::resources const& handle,
 {
   bool is_row_major = detail::is_row_major(y, z);
 
+  // WARNING: The following copy is working around a bug in cusparse which causes an alignment issue
+  // and incorrect results. This bug is fixed in CUDA 12.5+ so this workaround shouldn't be removed
+  // until that version is supported.
+  auto z_tmp = raft::make_device_matrix<ValueType, IndexType>(handle, z.extent(0), z.extent(1));
+  raft::copy(z_tmp.data_handle(), z.data_handle(), z.size(), raft::resource::get_cuda_stream(handle));
```

Comment:
Fixes style checker:

```suggestion
  raft::copy(
    z_tmp.data_handle(), z.data_handle(), z.size(), raft::resource::get_cuda_stream(handle));

```

Reply:
@mfoerste4's [PR 2185](https://github.com/rapidsai/raft/pull/2185/) uses an intermediate `device_uvector` instead of a `raft::device_matrix` because the matrix's `size()` isn't suitable for the copy here.

That PR is targeting `branch-24.02`. Should we merge that PR into 24.02 and rerelease RAFT, or should we merge that PR into this one and include it in 24.04?